docs/_posts/2024-07-18-Run-your-pandas-workloads-on-cluster.md

```diff
@@ -0,0 +1,152 @@
+
+Once the Ray cluster is up and running, set the RAY_ADDRESS environment variable to the Ray dashboard URL on your client machine.
+```bash
+export RAY_ADDRESS=URL_OF_RAY_DASHBOARD
+```
+Now, make minor changes to your Python script by importing Pandas from Modin and initializing Ray.
+
+```python
+import modin.pandas as pd
+import ray
+ray.init(address="auto", logging_level="WARNING")
+# your workload without any changes
+```
+
+
+
+You can view the complete Python script below.
+<details>
+  <summary>modin_workload.py</summary>
+
+    <pre><code class="language-python">
+    # import pandas as pd
+    import modin.pandas as pd
+    import ray
+    from modin.utils import execute # execute is just used to make sure all the asynchronous operations are complete as we benchmark.
+    ray.init(address="auto", logging_level="WARNING")
+    file_path = "/home/ray/data/big_yellow.csv"
+    df = pd.read_csv(file_path)
+    df = df.map(str)
+    result = pd.DataFrame()
+
+    payment_type_map = {
+        1: 'Credit Card',
+        2: 'Cash',
+        3: 'No Charge',
+        4: 'Dispute',
+        5: 'Unknown',
+        6: 'Voided Trip'
+    }
+
+    result["df_payment_type"] = df['payment_type'].map(payment_type_map)
+
+    def custom_function(row):
+        return str(row["passenger_count"]) + " passengers were picked up at " + str(row["tpep_pickup_datetime"])
+
+    result["description"] = df.apply(custom_function, axis=1)
+    execute(result)
+    </code></pre>
+
+  ```
+</details>
+
+
+### Results and discussion
+
+I observed that by increasing the number of cluster nodes, Modin could handle the workload effectively with 4 nodes or more. Significant performance improvements were evident as we scaled from 4 to 32 nodes. Given that Modin operates efficiently with 4 nodes and both Pandas and Modin struggle with fewer than 3 nodes, it suggests the workload demands nearly 4 times the memory (32GB) of a single t3.2xlarge instance.
+
+To provide a fair comparison of the performance gains offered by Modin over Pandas, we decided to benchmark the script on an instance with over 120 GB of memory. We chose an r6a.4xlarge instance, which boasts 128GiB of RAM.
+
+With Pandas, the script took almost 400 seconds to execute, which serves as our baseline.
+
+
+
+### Performance on a Single Node
+The graph reveals an interesting twist: Modin on a single node starts off slower than Pandas when using 4 workers or fewer. This could be because the overhead of distributing data and spinning up the Ray cluster doesn't pay off with so few workers. It's like trying to split chores among two people when you really need a full team. But here's where it gets exciting: crank up the CPU count (even just to 8), and Modin shifts into high gear, outperforming Pandas even on a single node. Talk about a comeback!
```

Comment:
```suggestion

The graph reveals an interesting twist: Modin on a single node starts off slower than pandas
when using 4 workers or fewer. This could be because the overhead of distributing data and spinning up
the Ray cluster doesn't pay off with so few workers. It's like trying to split chores among two people
when you really need a full team. But here's where it gets exciting: crank up the CPU count
(even just to 8), and Modin shifts into high gear, outperforming pandas even on a single node.
Talk about a comeback!
```